src/app/core/notifications/notifications_manager.nim

```diff
@@ -0,0 +1,170 @@
+import NimQml, json, chronicles
+
+import ../../global/app_signals
+import ../../global/global_singleton
+import ../eventemitter
+import details
+
+export details
+
+logScope:
+  topics = "notifications-manager"
+
+const NOTIFICATION_SOUND = "qrc:/imports/assets/audio/notification.wav"
+
+# Signals which may be emitted by this class:
+const SIGNAL_DISPLAY_APP_NOTIFICATION* = "displayAppNotification"
+const SIGNAL_OS_NOTIFICATION_CLICKED* = "osNotificationClicked"
+
+# Notification preferences
+const NOTIFY_ABOUT_ALL_MESSAGES = 0
+const NOTIFY_JUST_ABOUT_MENTIONS = 1
+const NOTIFY_NOTHING_ABOUT = 2
+
+# Anonymous preferences
+const PREVIEW_ANONYMOUS = 0
+const PREVIEW_NAME_ONLY = 1
+const PREVIEW_NAME_AND_MESSAGE = 2
+
+type
+  NotificationArgs* = ref object of Args
+    title*: string
+    message*: string
+    details*: NotificationDetails
+
+  ClickedNotificationArgs* = ref object of Args
+    details*: NotificationDetails
+
+QtObject:
+  type NotificationsManager* = ref object of QObject
+    events: EventEmitter
+    osNotification: StatusOSNotification
+    soundManager: StatusSoundManager
+
+  proc processNotification(self: NotificationsManager, title: string, message: string, details: NotificationDetails)
+
+  proc setup(self: NotificationsManager, events: EventEmitter) =
+    self.QObject.setup
+    self.events = events
+    self.osNotification = newStatusOSNotification()
+    self.soundManager = newStatusSoundManager()
+
+    signalConnect(self.osNotification, "notificationClicked(QString)", self, "onOSNotificationClicked(QString)", 2)
```

Comment:
@jrainville  it comes from the `NimQml` and that `signalConnect` is actually a real `Qt::connection` so connecting 2 `QObject`s.

> Could our own events bus have worked?

For what I used here this is the approach, I used Qt signal/slot mechanism for this.

> My only concern is having to put slot functions in string can cause issues no?

No, that's what `signalConnect` expected, take a look into `NimQml`.

Reply:
> No, that's what signalConnect expected, take a look into NimQml

Sorry my question wasn't clear enough. I meant could it cause issues in the case where the slot gets renamed or adds a new parameter? If the string that points to that slot isn't uptated, would it fail to compile or would it crash on execution? If it's a compilation error, then that's fine, since we can correct it.

Reply:
@jrainville whatever change one does there and if signal and slot do not match in their list of params or if signal/slot names are missed we will get that in a log as `QObject::connect` info with description of a real error (but the app will be compilable).

Reply:
Ah ok, as long as it doesn't crash, it's not too bad.